[site-isolation] Hit test for use of touch event listeners set on layers

nmoucht opened the pull request:
#### 6854c7c79ce374ba8629031315334e33c0b0f2c0
<pre>
[site-isolation] Hit test for use of touch event listeners set on layers
<a href="https://bugs.webkit.org/show_bug.cgi?id=288180">https://bugs.webkit.org/show_bug.cgi?id=288180</a>
<a href="https://rdar.apple.com/145271607">rdar://145271607</a>

Reviewed by Simon Fraser.

Hit test the layer tree in the UI process when handling touch events so that we can
get the corresponding event region set on the layer in the web process.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
(WebCore::ScrollingTree::commitTreeStateInternal):
* Source/WebCore/rendering/EventRegion.cpp:
(WebCore::EventRegion::eventTrackingTypeForPoint const):
(WebCore::EventRegion::eventListenerRegionTypesForPoint const):
* Source/WebCore/rendering/EventRegion.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::eventRegionForLayer):
(WebKit::layerEventRegionContainsPoint):
(WebKit::RemoteLayerTreeDrawingAreaProxy::eventTrackingTypeForPoint):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeFrameScrollingNodeRemoteIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeFrameScrollingNodeRemoteIOS.mm:
(WebKit::ScrollingTreeFrameScrollingNodeRemoteIOS::commitStateBeforeChildren):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):

Canonical link: <a href="https://commits.webkit.org/291134@main">https://commits.webkit.org/291134@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/529dce4c7dca010f3335342164c32800520d0861

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91932 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11455 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1007 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96868 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42539 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93982 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11812 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19945 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70546 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28028 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94933 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9007 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83271 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50873 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8737 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/866 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41753 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79059 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/866 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98897 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19055 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/14211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79574 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19307 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79117 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78800 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19538 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23339 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/647 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/12094 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19036 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24246 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18733 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22192 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20485 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/rendering/EventRegion.cpp:
```diff
@@ -630,18 +631,81 @@ void EventRegion::uniteEventListeners(const Region& region, OptionSet<EventListe
+        regionTypes.add(EventListenerRegionType::TouchStart);
+    if (m_touchEventListenerRegion.start.contains(point))
+        regionTypes.add(EventListenerRegionType::NonPassiveTouchStart);
+    if (m_nonPassiveTouchEventListenerRegion.move.contains(point))
```
Some blank lines would help here but I think it's wrong that your `m_nonPassiveTouchEventListenerRegion` and `EventListenerRegionType::NonPassiveTouchMove` are not paired.

nmoucht replied:
Fixed.

smfr commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -13809,16 +13812,6 @@
 			path = Automation;
 			sourceTree = "<group>";
 		};
-		99A5149C2D5C0CA000937177 /* protocol */ = {
```
Unrelated project change.

nmoucht replied:
Fixed.

smfr reviewed (approved):
Can we test this with some region dumping?

nmoucht committed:
[site-isolation] Hit test for use of touch event listeners set on layers
https://bugs.webkit.org/show_bug.cgi?id=288180
rdar://145271607

Reviewed by Simon Fraser.

Hit test the layer tree in the UI process when handling touch events so that we can
get the corresponding event region set on the layer in the web process.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
(WebCore::ScrollingTree::commitTreeStateInternal):
* Source/WebCore/rendering/EventRegion.cpp:
(WebCore::EventRegion::eventTrackingTypeForPoint const):
(WebCore::EventRegion::eventListenerRegionTypesForPoint const):
* Source/WebCore/rendering/EventRegion.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::eventRegionForLayer):
(WebKit::layerEventRegionContainsPoint):
(WebKit::RemoteLayerTreeDrawingAreaProxy::eventTrackingTypeForPoint):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeFrameScrollingNodeRemoteIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeFrameScrollingNodeRemoteIOS.mm:
(WebKit::ScrollingTreeFrameScrollingNodeRemoteIOS::commitStateBeforeChildren):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):

Canonical link: https://commits.webkit.org/291134@main
